server/game/cards/02_SHD/events/RelentlessPursuit.ts

```diff
@@ -0,0 +1,41 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, Trait, WildcardCardType, WildcardZoneName } from '../../../core/Constants';
+
+export default class RelentlessPursuit extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '5778949819',
+            internalName: 'relentless-pursuit',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose a friendly unit. If the friendly unit is a Bounty Hunter, give a Shield token to it.',
+            targetResolver: {
+                controller: RelativePlayer.Self,
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: WildcardZoneName.AnyArena,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (context) => context.target.hasSomeTrait(Trait.BountyHunter),
+                    onTrue: AbilityHelper.immediateEffects.giveShield(),
+                    onFalse: AbilityHelper.immediateEffects.noAction({ hasLegalTarget: true }),
+                })
+            },
+            then: (thenContext) => ({
+                title: 'It captures an enemy non-leader unit that costs the same as or less than it.',
+                targetResolver: {
```

Comment:
Yeah, my initial implementation was using two `targetResolvers` but I couldn't figure out how to make it work with the case where you want to select a unit that cannot capture anything (scenario 3 in the tests). Is there a way to make the target resolver optional but without the ability to choose no target?

Reply:
Ahh okay, makes sense. Yeah in jigoku / l5r they had it set up so that all dependent target resolvers must also have a target, but that game had stricter rules about effects needing to resolve so I think it might not apply for us. Just hadn't hit a scenario yet where it mattered.

I think that should be relatively straightforward to fix but I might be wrong; it's worth giving it a shot. Do you mind trying to remove the requirement that dependent targets have to resolve and see if it breaks any other tests? Can just look at the usages of `dependentTarget` in TargetResolver.ts and its derived classes.

If that ends up going down a rabbit hole of broken tests and you don't feel like dealing with it just lmk and I can carve out some time to untangle it. But definitely we should get this fixed one way or another

Reply:
Removing the logic completely was breaking 3 tests (e.g. you were now required to select a friendly unit when playing Strike True even if there are no enemy units) so I've just disabled it in the case the target resolver ha an immediate effect.